modules/homebrew.nix

```diff
@@ -1,219 +1,751 @@
+        '';
+      };
+
+      brewfileLine = mkInternalOption { type = types.nullOr types.str; };
+    };
+
+    config =
+      let
+        sCfg = mkProcessedSubmodConfig config;
+        sCfgSubset = removeAttrs sCfg [ "name" "restart_service" ];
+      in
+      {
+        brewfileLine =
+          "brew ${sCfg.name}"
+          + optionalString (sCfgSubset != { }) ", ${mkBrewfileLineOptionsListString sCfgSubset}"
+          # We need to handle the `restart_service` option seperately since it can be either bool
+          # or the string value ":changed" the Brewfile.
+          + optionalString (sCfg ? restart_service) (
+            ", restart_service: " + (
+              if isBool config.restart_service then sCfg.restart_service
+              else ":${config.restart_service}"
+            )
+          );
+      };
+  };
+
+  caskOptions = { config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = "The name of the cask to install.";
+      };
+      args = mkOption {
+        # We use this type rather then `types.nullOr (types.submodule caskArgsOptions)` so that all
+        # the documentation about cask args aren't included a second time.
+        type = with types; either (enum [ null ]) (types.submodule caskArgsOptions);
+        default = null;
+        description = ''
+          Arguments passed to <command>brew install --cask</command> when installing this cask. See
+          ${mkDocOptionLink "homebrew.caskArgs"} for the available options.
+        '';
+      };
+      greedy = mkNullOrBoolOption {
+        description = ''
+          Whether to always upgrade this cask regardless of whether it's unversioned or it updates
+          itself.
+        '';
+      };
+
+      brewfileLine = mkInternalOption { type = types.nullOr types.str; };
+    };
+
+    config =
+      let
+        sCfg = mkProcessedSubmodConfig config;
+        sCfgSubset = removeAttrs sCfg [ "name" ];
+      in
+      {
+        brewfileLine =
+          "cask ${sCfg.name}"
+          + optionalString (sCfgSubset != { }) ", ${mkBrewfileLineOptionsListString sCfgSubset}";
+      };
+  };
 in
 
 {
+  # Interface --------------------------------------------------------------------------------------
+
+  imports = [
+    (mkRenamedOptionModule [ "homebrew" "autoUpdate" ] [ "homebrew" "onActivation" "update" ])
+    (mkRenamedOptionModule [ "homebrew" "cleanup" ] [ "homebrew" "onActivation" "cleanup" ])
+  ];
+
   options.homebrew = {
     enable = mkEnableOption ''
-      configuring your Brewfile, and installing/updating the formulas therein via
-      the <command>brew bundle</command> command, using <command>nix-darwin</command>.
-
-      Note that enabling this option does not install Homebrew. See the Homebrew website for
-      installation instructions: https://brew.sh
+      <command>nix-darwin</command> to manage installing/updating/upgrading Homebrew taps, formulae,
```

Comment:
This sentence seems to start a bit non sequitur, "nix-darwin to manage..."

Reply:
`mkEnableOption` creates the description like this: `"Whether to enable ${<the argument to mkEnableOption>}."`.